### PR TITLE
Kyle adds

### PR DESCRIPTION
**Pull request by kegillett**
Fixed some spelling errors and added a problem to the whiteboarding folder

### CODE REVIEW

File: whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Here's a start...10-28-18
+
+str1 = abcdefghijklmnopqrstuvxyz
+str2 = 123456789
+
+function(str1, str2){
```
Comment (ed42311) [inline whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js]: Syntax error when run in `node`
```/Users/edwardweymouth/code/practice/keg-middle.js:4
 function(str1, str2){
         ^

SyntaxError: Unexpected token (```

Why?

File: whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Here's a start...10-28-18
+
+str1 = abcdefghijklmnopqrstuvxyz
```
Comment (ed42311) [inline whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js]: It probably would have error out here, but hoisting, ya know?

Let's check it out, yup -> `ReferenceError: abcdefghijklmnopqrstuvxyz is not defined`

File: whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+str2 = 123456789
+
+function(str1, str2){
+  let l2 = str2.length
+  let l1 = str1.length
+  let a = str1.splice(0, x)
```
Comment (ed42311) [inline whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js]:
```/Users/edwardweymouth/code/practice/keg-middle.js:7
  let a = str1.splice(0, x)
                         ^

ReferenceError: x is not defined
```

File: whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Here's a start...10-28-18
+
+str1 = abcdefghijklmnopqrstuvxyz
+str2 = 123456789
```
Comment (ed42311) [inline whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js]: Both these variables are set without var, let, or const, making them globals

File: whiteBoarding/squareDigits/keg-squareDigits.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+function squareDigits(num) {
```
Comment (ed42311) [inline whiteBoarding/squareDigits/keg-squareDigits.js]: Interesting solution and done without an array.  Is there another way to do this with higher order functions?

What are the pros and cons of using the higher order functions versus a simple for loop?

File: answers/general.md [hunk 6]
```diff
@@ -5,3 +5,9 @@
 
 * Asynchronous code
  - code in which all processes are able to work independently of each other without waiting for other process to finish before more start.
+
+* Immutability
+ - means literally that nothing can be mutated. This is the difference between const (which is immutable) and let or var (which are mutable). In javascript only primitive types can really be immutable (string, number, bool). For objects, they are mutable by default b/c the assignment operator is essentially just a memory pointer, so any change will effect all.
```
Comment (ed42311) [inline answers/general.md]: means -> **M**eans

File: answers/general.md [hunk 7]
```diff
@@ -5,3 +5,9 @@
 
 * Asynchronous code
  - code in which all processes are able to work independently of each other without waiting for other process to finish before more start.
```
Comment (ed42311) [inline answers/general.md]: code -> **C**ode

File: answers/general.md (hunk 6 above)
Comment (kegillett) [reply answers/general.md]: Fixed

File: answers/general.md (hunk 7 above)
Comment (kegillett) [reply answers/general.md]: Fixed

File: whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js (hunk 1 above)
Comment (kegillett) [reply whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js]: b/c I didn't give the function a name.

File: whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js (hunk 2 above)
Comment (kegillett) [reply whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js]: needs to actually be a string

File: whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js (hunk 3 above)
Comment (kegillett) [reply whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js]: changed variable names in only one place...

File: whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js (hunk 4 above)
Comment (kegillett) [reply whiteBoarding/splitMiddleStr/keg-splitMiddleStr.js]: sure are

File: whiteBoarding/squareDigits/keg-squareDigits.js (hunk 5 above)
Comment (kegillett) [reply whiteBoarding/squareDigits/keg-squareDigits.js]: It uses an array to store the results. You could do it all in line with toString().split('').map(with calculations in here).join('') probably. Not sure of advantages other than less code.

File: answers/general.md [hunk 8]
```diff
@@ -4,4 +4,10 @@
...
 
 * Asynchronous code
- - code in which all processes are able to work independently of each other without waiting for other process to finish before more start.
+ - Code in which all processes are able to work independently of each other without waiting for other process to finish before more start.
+
+* Immutability
+ - Means literally that nothing can be mutated. This is the difference between const (which is immutable) and let or var (which are mutable). In javascript only primitive types can really be immutable (string, number, bool). For objects, they are mutable by default b/c the assignment operator is essentially just a memory pointer, so any change will effect all.
+
+* Attributes vs Property
+- Properties can be thought of as children, or maybe siblings of attributes. When you define attributes in you html and once the browser parses it, a corresponding DOM node is created. This node is an object, and therefore it has properties. They often mirror each other, except like in cases like inputs where the .value returns the input and the .getAttribute('value') returns the default value placeholder text.
```
Comment (ed42311) [inline answers/general.md]: ...in you**r** html and...
...like in cases ?**such as**? inputs

File: whiteBoarding/squareDigits/keg-squareDigits.js (hunk 5 above)
Comment (ed42311) [reply whiteBoarding/squareDigits/keg-squareDigits.js]: That is an advantage, what is the disadvantage.  Write the code out in a similar function and run it.

**Pull request merged**